microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/integration/config/CustomMetricModelPropsTest.java

```diff
@@ -0,0 +1,74 @@
+package org.jboss.eap.qe.microprofile.metrics.integration.config;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import java.io.IOException;
+import java.util.concurrent.TimeoutException;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.arquillian.api.ServerSetupTask;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.client.helpers.ClientConstants;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.dmr.ModelNode;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.ConfigurationException;
+import org.jboss.eap.qe.microprofile.tooling.server.configuration.creaper.ManagementClientProvider;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.runner.RunWith;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup(CustomMetricModelPropsTest.SetupTask.class)
+public class CustomMetricModelPropsTest extends CustomMetricBaseTest {
+
+    private static final PathAddress CONFIG_SOURCE_PROPS_ADDRESS = PathAddress.pathAddress()
+            .append(SUBSYSTEM, "microprofile-config-smallrye")
+            .append("config-source", "props");
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap.create(WebArchive.class, CustomMetricModelPropsTest.class.getSimpleName() + ".war")
+                .addClasses(CustomCounterIncrementProvider.class, CustomCounterMetric.class, CustomMetricService.class,
+                        CustomMetricApplication.class, CustomMetricAppInitializer.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return webArchive;
+    }
+
+    void setConfigProperties(int increment) throws IOException, ConfigurationException, TimeoutException, InterruptedException {
+        ModelNode properties = new ModelNode().add(INCREMENT_CONFIG_PROPERTY, increment);
+        Assert.assertEquals(ClientConstants.SUCCESS, managementClient.getControllerClient()
+                .execute(Util.getWriteAttributeOperation(CONFIG_SOURCE_PROPS_ADDRESS, "properties", properties))
+                .get(ClientConstants.OUTCOME)
+                .asString());
+        new Administration(ManagementClientProvider.onlineStandalone()).reload();
+    }
+
+    public static class SetupTask implements ServerSetupTask {
```

Comment:
Please describe purpose of this SetupTask.

Reply:
SetupTask is specific to the scenario. `setup` method has few self-descriptive lines.

Reply:
To grasp what it does I have to read those line and also see one additional variable. I now guess that it add a file as a parameter somewhere. But this took me several minutes instead of reading a sentence describing this process.

Reply:
Fixing...